packages/node/src/integrations/contextlines.ts

```diff
@@ -0,0 +1,141 @@
+import { getCurrentHub } from '@sentry/core';
+import { Event, EventProcessor, Integration } from '@sentry/types';
+import { addContextToFrame } from '@sentry/utils';
+import { readFile } from 'fs';
+import { LRUMap } from 'lru_map';
+
+import { NodeClient } from '../client';
+
+const FILE_CONTENT_CACHE = new LRUMap<string, string | null>(100);
+const DEFAULT_LINES_OF_CONTEXT = 7;
+
+function readTextFileAsync(path: string): Promise<string> {
+  return new Promise((resolve, reject) => {
+    readFile(path, 'utf8', (err, data) => {
+      if (err) reject(err);
+      else resolve(data);
+    });
+  });
+}
+
+/**
+ * Resets the file cache. Exists for testing purposes.
+ * @hidden
+ */
+export function resetFileContentCache(): void {
+  FILE_CONTENT_CACHE.clear();
+}
+
+interface ContextLinesOptions {
+  /**
+   * Sets the number of context lines for each frame when loading a file
+   *
+   * Set to 0 to disable loading and inclusion of source files.
+   * */
+  frameContextLines?: number;
+}
+
+/** Add node modules / packages to the event */
+export class ContextLines implements Integration {
+  /**
+   * @inheritDoc
+   */
+  public static id: string = 'ContextLines';
+
+  /**
+   * @inheritDoc
+   */
+  public name: string = ContextLines.id;
+
+  public constructor(private readonly _options: ContextLinesOptions = {}) {}
+
+  /**
+   * @inheritDoc
+   */
+  public setupOnce(addGlobalEventProcessor: (callback: EventProcessor) => void): void {
+    if (this._options.frameContextLines == undefined) {
+      const initOptions = getCurrentHub().getClient<NodeClient>()?.getOptions();
+      // eslint-disable-next-line deprecation/deprecation
+      this._options.frameContextLines = initOptions?.frameContextLines;
+    }
+
+    addGlobalEventProcessor(event => this.addToEvent(event));
+  }
+
+  /** Processes an event and adds context lines */
+  public async addToEvent(event: Event): Promise<Event> {
+    const contextLines =
+      this._options.frameContextLines !== undefined ? this._options.frameContextLines : DEFAULT_LINES_OF_CONTEXT;
```

Comment:
Is there a reason to do this here rather than in `setupOnce`? I don't think the value is going to change, will it?